packages/website/components/account/currentBillingPlanCard/currentBillingPlanCard.js

```diff
@@ -1,27 +1,30 @@
+import { formatAsStorageAmount, formatCurrency } from '../../../lib/utils';
+
 const CurrentBillingPlanCard = ({ plan }) => {
   return (
     <div className="billing-card card-transparent">
       {plan !== undefined && (
         <div key={plan.title} className="billing-plan">
           <div className="billing-plan-overview">
-            <h4 className="billing-plan-title">{plan.title}</h4>
-            <div className="billing-plan-amount">{plan.price}</div>
+            <h4 className="billing-plan-title">{plan.metadata['UI Label']}</h4>
```

Comment:
Will you encapsulate the stripe metadata processing inside `StripeSubscriptionService`? More here https://github.com/web3-storage/web3.storage/pull/2152/files#r1065213416

If we do that, then:
1) this frontend code has no stripe specifics, it only depends on billing-types types
2) this code won't have errors like:
    * If someone sets a customer to a custom price, but sets metadata 'UI label' instead of 'UI Label', the end-user will see 'undefined' here
    * if someone sets a customer to a custom price with no tiers, the end-user will see '/mo'

Reply:
Ideally whenever someone changes a 'custom price' in stripe (including metadata), we either:
* get a big loud error via sentry as soon as `StripeSubscriptionService` encounters it an tries to parse the price into a `StoragePrice`
* get no error and feel confident that everyone consuming the StoragePrice has everything they will need (i.e. no need here to handle case where `tiers` is an empty array by putting optional chaining operators in here, so need for this ui code to handle `NaN` currencies in formatCurrency)

Reply:
> get no error and feel confident that everyone consuming the StoragePrice has everything they will need

I was opting for this one.  

That's related to my original rationale behind not doing any additional mapping.  This metadata doesn't pertain to any business rules and really only relates to view logic and text.  Our API, in my original opinion, was kept more flexible if staying agnostic to what's actually kept in this metadata.  Mapping this data seemed like unnecessary obfuscation, and if we switch to another billing provider later then presumably they'd offer a similar way to store this sort of metadata which could _then_ be mapped to the shape of Stripe's metadata (i.e., it seemed premature to map it to our own interface).

But that said, I also agree with what you're saying and so I wrote it that way as well (or similar to -- let me know if there's anything I missed in your intent), and it does make things tidier in the UI.

Reply:
>  This metadata doesn't pertain to any business rules and really only relates to view logic and text.

If that's true, then one might argue the website-specific view data shouldn't be stored in stripe metadata. Otherwise someone may come along and read/write from that metadata without knowing it's tightly coupled to the website frontend requirements.
I think it'd be ideal if we treat data in stripe *as business data* and not just view-related, but I hear that wasn't your intent, though, and that you prefer to do it another way.